third_party/terraform/resources/resource_storage_transfer_job.go

```diff
@@ -0,0 +1,728 @@
+				ValidateFunc: validation.IntBetween(0, 24),
+			},
+			"minutes": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(0, 59),
+			},
+			"seconds": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(0, 60),
+			},
+			"nanos": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(0, 999999999),
+			},
+		},
+	}
+}
+
+func dateObjectSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"year": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(0, 9999),
+			},
+
+			"month": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(1, 12),
+			},
+
+			"day": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(0, 31),
+			},
+		},
+	}
+}
+
+func gcsDataSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"bucket_name": {
+				Required: true,
+				Type:     schema.TypeString,
+			},
+		},
+	}
+}
+
+func awsS3DataSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"bucket_name": {
+				Required: true,
+				Type:     schema.TypeString,
+			},
+			"aws_access_key": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"access_key_id": {
+							Type:      schema.TypeString,
+							Required:  true,
+							Sensitive: true,
+						},
+						"secret_access_key": {
+							Type:      schema.TypeString,
+							Required:  true,
+							Sensitive: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func httpDataSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"list_url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceStorageTransferJobCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	transferJob := &storagetransfer.TransferJob{
+		Description:  d.Get("description").(string),
+		ProjectId:    project,
+		Status:       d.Get("status").(string),
+		Schedule:     expandTransferSchedules(d.Get("schedule").([]interface{})),
+		TransferSpec: expandTransferSpecs(d.Get("transfer_spec").([]interface{})),
+	}
+
+	var res *storagetransfer.TransferJob
+
+	err = retry(func() error {
+		res, err = config.clientStorageTransfer.TransferJobs.Create(transferJob).Do()
+		return err
+	})
+
+	if err != nil {
+		fmt.Printf("Error creating transfer job %v: %v", transferJob, err)
+		return err
+	}
+
```

Comment:
We should `SetId` in `Create` after a successful create; that way, if `Read` fails afterwards (such as to a `503`) Terraform will know we've created the resource and won't try recreating it.